contrib/database/sql/metrics.go

```diff
@@ -33,20 +33,20 @@ var interval = 10 * time.Second
 
 // pollDBStats calls (*DB).Stats on the db at a predetermined interval. It pushes the DBStats off to the statsd client.
 // the caller should always ensure that db & statsd are non-nil
-func pollDBStats(statsd internal.StatsdClient, db *sql.DB) {
+func pollDBStats(statsd internal.StatsdClient, db *sql.DB, tags []string) {
 	log.Debug("DB stats will be gathered and sent every %v.", interval)
 	for range time.NewTicker(interval).C {
 		log.Debug("Reporting DB.Stats metrics...")
 		stat := db.Stats()
-		statsd.Gauge(MaxOpenConnections, float64(stat.MaxOpenConnections), []string{}, 1)
-		statsd.Gauge(OpenConnections, float64(stat.OpenConnections), []string{}, 1)
-		statsd.Gauge(InUse, float64(stat.InUse), []string{}, 1)
-		statsd.Gauge(Idle, float64(stat.Idle), []string{}, 1)
-		statsd.Gauge(WaitCount, float64(stat.WaitCount), []string{}, 1)
-		statsd.Timing(WaitDuration, stat.WaitDuration, []string{}, 1)
-		statsd.Gauge(MaxIdleClosed, float64(stat.MaxIdleClosed), []string{}, 1)
-		statsd.Gauge(MaxIdleTimeClosed, float64(stat.MaxIdleTimeClosed), []string{}, 1)
-		statsd.Gauge(MaxLifetimeClosed, float64(stat.MaxLifetimeClosed), []string{}, 1)
+		statsd.Gauge(MaxOpenConnections, float64(stat.MaxOpenConnections), tags, 1)
+		statsd.Gauge(OpenConnections, float64(stat.OpenConnections), tags, 1)
+		statsd.Gauge(InUse, float64(stat.InUse), tags, 1)
+		statsd.Gauge(Idle, float64(stat.Idle), tags, 1)
+		statsd.Gauge(WaitCount, float64(stat.WaitCount), tags, 1)
+		statsd.Timing(WaitDuration, stat.WaitDuration, tags, 1)
+		statsd.Gauge(MaxIdleClosed, float64(stat.MaxIdleClosed), tags, 1)
+		statsd.Gauge(MaxIdleTimeClosed, float64(stat.MaxIdleTimeClosed), tags, 1)
+		statsd.Gauge(MaxLifetimeClosed, float64(stat.MaxLifetimeClosed), tags, 1)
```

Comment:
@mtoffl01 Was there any reason behind not allowing tags from the beginning?

Checking the code, I see that for the purpose of this PR, it should be enough to set `DD_TAGS` or `OTEL_PROPAGATORS` (if using OTel) to set these global tags.

WDYT @KaibutsuX?

Reply:
I don't have enough context or documentation to know what `DD_TAGS` is or how I would set it as an end-user.

Reply:
`DD_TAGS` is properly explained in [our Go tracer public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/go/).

We understand this should be enough to achieve what you are trying with this PR.

Reply:
That links says: "A list of default tags to be added to every span and profile." From my tracing through the code, I could not see that this same list of user-supplied tags was included in these metrics used in `pollDBStats`

Reply:
Yes, we understand it may not that be clear from the code. Sorry about that. It inherits the global tags from the tracer config, set by [`globalconfig.SetStatsTags`](https://github.com/DataDog/dd-trace-go/blob/471d72351b4ef7ed5cd69efddebaa3a3ccd4b3e2/ddtrace/tracer/option.go#L836), which is initialized by [`newConfig`](https://github.com/DataDog/dd-trace-go/blob/471d72351b4ef7ed5cd69efddebaa3a3ccd4b3e2/ddtrace/tracer/option.go#L412), and retrieved when [setting up the DB statsd client](https://github.com/DataDog/dd-trace-go/blob/471d72351b4ef7ed5cd69efddebaa3a3ccd4b3e2/contrib/database/sql/option.go#L43).

We'll update the public documentation to make sure it reflects this detail.